lib/sles4sap_publiccloud.pm

```diff
@@ -999,6 +1000,91 @@ sub saphanasr_showAttr_version {
     }
 }
 
+=head2 wait_for_cluster
+
+    Verifies that nodes are online, resources are started and DB is in sync
+
+=cut
+
+sub wait_for_cluster {
+    my ($self, $wait_time, $max_retries) = @_;
+
+    $wait_time //= 10;
+    $max_retries //= 7;
+
+    while ($max_retries > 0) {
+        my $hanasr_output = $self->run_cmd(cmd => 'SAPHanaSR-showAttr', quiet => 1);
+        my $crm_output = $self->run_cmd(cmd => 'crm_mon -r -R -n -N -1', quiet => 1);
+
+        my @lines = split /\n/, $hanasr_output;
+
+        # Look for the line that starts with "Hosts"
+        my $start_idx = 0;
+        for (my $i = 0; $i < @lines; $i++) {
+            if ($lines[$i] =~ /^\s*Hosts\s+/) {
+                $start_idx = $i;
+                last;
+            }
+        }
+
+        # Only get lines after here
+        @lines = @lines[$start_idx..$#lines];
+
+        # Get rif of header
+        shift @lines;
```

Comment:
why not to get rid also of next line that is just `----------------`. In any case later this function will skip it with `next if` at line 1054